archetype/engine-v2/src/main/java/io/helidon/build/archetype/engine/v2/context/ContextSerializer.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.build.archetype.engine.v2.context;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * Context serializer.
+ */
+public class ContextSerializer implements ContextEdge.Visitor {
+
+    private static final Set<ContextValue.ValueKind> DEFAULT_VALUE_KIND_FILTER =
+            Set.of(ContextValue.ValueKind.EXTERNAL, ContextValue.ValueKind.USER);
+    private static final Predicate<ContextEdge> DEFAULT_CONTEXT_FILTER =
```

Comment:
Both `DEFAULT_CONTEXT_FILTER` and `DEFAULT_VALUE_MAPPER` can be `private static` methods used with the method reference operator `::`.